File: src/main/java/com/jack/Student/Student.java
package com.jack.Student;

public class Student {
    String name;
    String id;
    int english;
    int math;


    public Student(String name, int math, int english) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public int highest(){
        return (english > math) ? english : math;
    }

    public void print(){
        System.out.print(name + "\t" + english + "\t" +
                math + "\t" + getAverage());
        if(getAverage() >= 60){
            System.out.println("\tPass");
        } else {
            System.out.println("\tFailed");
        }
    }

    public int getAverage(){
        return (math + english) / 2;
    }
}

File: src/main/kotlin/com/jack/Student/Student.kt
package com.jack.kotlin

import java.util.*

fun main() {
    //userInput()
    val stu = Student("Jack", 77, 99)
    stu.print()
    println("Highest score : ${stu.highest()}")
}


class Student(var name : String, var english : Int, var math : Int){
    fun print(){
        println("$name\t$english\t$math\t${getAverage()}\t${passOrFailed()}\t${grading()}")
    }

    fun passOrFailed() = if(getAverage() >= 60) "Pass" else "Failed"

    fun grading() = when(getAverage()){
            in 90..100 -> 'A'
            in 80..89 -> 'B'
            in 70..79 -> 'C'
            in 60..69 -> 'D'
            else -> 'F'
    }

    fun getAverage() =  (english+math) / 2

    fun highest() = if(english > math){
        println("English")
        english
    } else {
        println("Math")
        math
    }
}

private fun userInput() {
    val scanner = Scanner(System.`in`)
    print("Please enter your name:")
    var name = scanner.next()
    print("Please enter your english:")
    var english = scanner.nextInt()
    print("Please enter your name:")
    var math = scanner.nextInt()
    val stu = Student(name, english, math)
    stu.print()
}

